roles/openshift_router/tasks/main.yml

```diff
@@ -5,7 +5,8 @@
 - name: Deploy OpenShift Router
   command: >
     {{ openshift.common.admin_binary }} router
-    --create
-    --credentials={{ openshift_master_config_dir }}/openshift-router.kubeconfig {{ _ortr_images|default() }}
+    --create --replicas={{ num_infra }}
+    --service-account=router
+    --credentials={{ openshift_master_config_dir }}/openshift-router.kubeconfig {{ _ortr_images|default("--type='haproxy-router'") }}
```

Comment:
Setting the default to `--type='haproxy-router'` seems odd to me, that's the default in the product anyway. I'd just default it to null like you've done with the router.